480.14nm

![480.14](480-setting.png)

A line drawn from the 500 pixel position has a brightest point at around pixel 920

490.14nm

![image2](490-setting.png)

A line drawn from the 500 pixel position on the vertical axis has a peak at approximately pixel 492

500.14nm

![image1](500-setting.png)

This line has a peak along the 500-pixel (vertical scale) at approximately 60

The points are therefore (60, 500.14), (492, 490.14), and (920, 480.14)

In [3]:
import numpy as np

In [4]:
pixels = np.array([60, 492, 920])
wavelengths = np.array([500.14, 490.14, 480.14])

In [22]:
x = np.vstack([np.atleast_2d(pixels), np.ones_like(pixels)]).T
y = np.atleast_2d(wavelengths).T
a = np.linalg.inv(x.T @ x) @ x.T @ y
print(a)

[[-2.32556463e-02]
 [ 5.01550770e+02]]


Based on the scale of the images, assume that the error in pixel number is $\pm 25$. This gives an error of

In [28]:
s = (np.sum((y - x.dot(a))**2) / 1)**0.5

The error in the slope is

In [33]:
s * np.sqrt(3 / (3 * np.sum(pixels**2) - np.sum(pixels)**2))

6.244955174532996e-05

and the error in the intercept

In [34]:
s * np.sqrt((np.sum(pixels**2)) / (3 * np.sum(pixels**2) - np.sum(pixels)**2))

0.037678425254528226

The scale for these plots is therefore $0.0232 \pm 6.24 \times 10^{-5}$ nm per pixel